src/pray.c

```diff
@@ -3946,66 +3946,235 @@ aligntyp alignment;
+				else if (uleft && not_fully_identified(uleft)) identify(uleft);
+				else if (uright && not_fully_identified(uright)) identify(uright);
+				else if (uarmc && not_fully_identified(uarmc)) identify(uarmc);
+				else if (uarm && not_fully_identified(uarm)) identify(uarm);
+				else if (uarmu && not_fully_identified(uarmu)) identify(uarmu);
+				else if (uarmh && not_fully_identified(uarmh)) identify(uarmh);
+				else if (uarmg && not_fully_identified(uarmg)) identify(uarmf);
+				else if (uarmf && not_fully_identified(uarmf)) identify(uarmf);
+				else if (uarms && not_fully_identified(uarms)) identify(uarms);
+				else {
+					for(otmp=invent; otmp; otmp=otmp->nobj)
+					if (not_fully_identified(otmp)) identify(otmp);
 				}
-			case 2: /* randomly identify items in the backpack */
-			case 3: /* do magic mapping */
-			case 4: /* give some food */
-			case 5: /* randomly bless items */
-		    /* weapon takes precedence if it interferes
-		       with taking off a ring or shield */
-
-		    if (uwep && !uwep->blessed) /* weapon */
-			    otmp = uwep;
-		    else if (uswapwep && !uswapwep->blessed) /* secondary weapon */
-			    otmp = uswapwep;
-		    /* gloves come next, due to rings */
-		    else if (uarmg && !uarmg->blessed)    /* gloves */
-			    otmp = uarmg;
-		    /* then shield due to two handed weapons and spells */
-		    else if (uarms && !uarms->blessed)    /* shield */
-			    otmp = uarms;
-		    /* then cloak due to body armor */
-		    else if (uarmc && !uarmc->blessed)    /* cloak */
-			    otmp = uarmc;
-		    else if (uarm && !uarm->blessed)      /* armor */
-			    otmp = uarm;
-		    else if (uarmh && !uarmh->blessed)    /* helmet */
-			    otmp = uarmh;
-		    else if (uarmf && !uarmf->blessed)    /* boots */
-			    otmp = uarmf;
-//ifdef TOURIST
-		    else if (uarmu && !uarmu->blessed)    /* shirt */
-			    otmp = uarmu;
-//endif
-		    /* (perhaps amulet should take precedence over rings?) */
-		    else if (uleft && !uleft->blessed)
-			    otmp = uleft;
-		    else if (uright && !uright->blessed)
-			    otmp = uright;
-		    else if (uamul && !uamul->blessed) /* amulet */
-			    otmp = uamul;
-		    else {
-			    for(otmp=invent; otmp; otmp=otmp->nobj)
-				if (!otmp->blessed)
+				break;
+			case 3: // give an intrinsic for 500-1500 turns, first of pois/slee/fire/cold/shock
+				timeout = rn1(1000, 500);
+
+				if(!(HPoison_resistance & FROMOUTSIDE)) {
+					You_feel(Poison_resistance ? "especially healthy." : "healthy.");
+					HPoison_resistance |= FROMOUTSIDE;
 					break;
-			    return; /* Nothing to do! */
-		    }
-		    bless(otmp);
-		    otmp->bknown = TRUE;
-		    if (!Blind)
-			    Your("%s %s.",
-				 what ? what :
-				 (const char *)aobjnam (otmp, "softly glow"),
-				 hcolor(NH_AMBER));
-			break;
+				}
+				
+				if(!(HSleep_resistance)) {
+					You_feel("wide awake.");				
+					if((HSleep_resistance & timeout) < TIMEOUT) {
```

Comment:
You already checked `!(HSleep_resistance)`. 
Instead of:
```
					if((HSleep_resistance & timeout) < TIMEOUT) {
					long timer = (HSleep_resistance & TIMEOUT) + timeout;
						HSleep_resistance &= ~TIMEOUT; //wipe old timer, leaving higher bits in place
						HSleep_resistance |= timer; //set new timer
						break;
					} else {
						HSleep_resistance |= timeout;
						break;
					}
```
Do:
```
HSleep_resistance |= timeout;
break;
```

And similar for the other resistances.

Reply:
Intentional, I intend to unify the intrinsic-giving code into a single function, right now all code that does that uses ugly copy-pastes from eat.c.